File: express-ts/src/modules/auth/auth.service.ts
import { UserRepositoryInterface , userRepository} from "../user/user.repository"
import { ApiError } from "../../utils/err.utils"
import { generateToken } from "../../utils/jwt.utils"

export interface AuthServiceInterface {
  LoginUser(email: string, password: string): string ; 
}

export class AuthService implements AuthServiceInterface {
  constructor (private userRepository: UserRepositoryInterface) {}

  LoginUser(email: string, password: string): string  {
    console.log(email, password)
    const user = this.userRepository.getUserByEmail(email)
    if (!user) {
      throw new ApiError(404, "User Not Found")
    }
    const compare = user.password === password
    if (!compare) {
      throw new ApiError(401, "Invalid email or password")
    }
    const token = generateToken({ id: user.id, email: user.email })
    return token
  }
}

export const authService = new AuthService(userRepository)

File: gofiber/internal/domain/auth/auth_service.go
package auth

import (
	"github.com/risoluzumaki/jwt/go/internal/domain/user"
	appErr "github.com/risoluzumaki/jwt/go/pkg/app"
	appUtils "github.com/risoluzumaki/jwt/go/pkg/utils"
)

type AuthServiceInterface interface {
	LoginUser(email string, password string) (string, error)
}

type AuthService struct {
	repo user.UserRepositoryInterface
}

func NewAuthService(ur user.UserRepositoryInterface) AuthServiceInterface {
	return &AuthService{
		repo: ur,
	}
}

func (s *AuthService) LoginUser(email string, password string) (string, error) {
	result, err := s.repo.GetByEmail(email)

	if err != nil {
		return "", appErr.NewError(404, "user not found")
	}

	if result.Password != password {
		return "", appErr.NewError(400, "Invalid Password")
	}

	token, err := appUtils.GenerateToken(result.ID, result.Email)
	if err != nil {
		return "", appErr.NewError(500, "Internal Server Error")
	}

	return token, nil
}
